django-stubs/contrib/auth/models.pyi

```diff
@@ -41,7 +41,7 @@ class PermissionsMixin(models.Model):
     user_permissions: models.ManyToManyField = models.ManyToManyField(Permission)
     def get_group_permissions(self, obj: None = ...) -> Set[str]: ...
     def get_all_permissions(self, obj: Optional[str] = ...) -> Set[str]: ...
-    def has_perm(self, perm: Union[Tuple[str, Any], str], obj: Optional[str] = ...) -> bool: ...
+    def has_perm(self, perm: str, obj: Optional[str] = ...) -> bool: ...
     def has_perms(self, perm_list: Union[List[str], Set[str], Tuple[str]], obj: None = ...) -> bool: ...
```

Comment:
Could you also make `has_perms` first parameter a `Collection[str]` and cleanup all those `obj: None` that are around?

Reply:
I've done the `Collection[str]` change but I'm not sure what kind of cleanup you want for the `obj: None`.